app/viewmodels/SecuritiesRequestedStatementsViewModel.scala

```diff
@@ -0,0 +1,142 @@
+                                                  renderMissingDocumentsGuidance: HtmlFormat.Appendable)
+
+object SecuritiesRequestedStatementsViewModel {
+
+  def apply(securityStatementsForEori: Seq[SecurityStatementsForEori])
+           (implicit messages: Messages): SecuritiesRequestedStatementsViewModel = {
+
+    val preparedStatementRows = prepareStatementRows(securityStatementsForEori)
+    val hasStatements = hasSecurityStatementsForEori(securityStatementsForEori)
+    val eoriHeading = preparedStatementRows.headOption.flatMap(renderEoriHeading).getOrElse(HtmlFormat.empty)
+    val pdfLink = preparedStatementRows.headOption.map(row => renderPdfLink(row.pdf))
+
+    SecuritiesRequestedStatementsViewModel(
+      statementRows = preparedStatementRows,
+      hasStatements = hasStatements,
+      renderEoriHeading = eoriHeading,
+      renderPdfLink = pdfLink,
+      renderMissingDocumentsGuidance = renderMissingDocumentsGuidance
+    )
+  }
+
+  def prepareStatementRows(securityStatements: Seq[SecurityStatementsForEori])
+                          (implicit messages: Messages): Seq[StatementRow] = {
+
+    for {
+      (statement, statementIndex) <- securityStatements.zipWithIndex
+      (requestedStatement, requestedIndex) <- statement.requestedStatements.sorted.zipWithIndex
+    } yield createStatementRow(statement, statementIndex, requestedStatement, requestedIndex)
+  }
+
+  private def createStatementRow(statement: SecurityStatementsForEori,
+                                 statementIndex: Int,
+                                 requestedStatement: SecurityStatementsByPeriod,
+                                 requestedIndex: Int)
+                                (implicit messages: Messages): StatementRow = {
+
+    val eori = if (statementIndex > 0) Some(statement.eoriHistory.eori) else None
+
+    val date = messages("cf.security-statements.requested.period",
+      dateAsDayMonthAndYear(requestedStatement.startDate),
+      dateAsDayMonthAndYear(requestedStatement.endDate))
+
+    val pdf = requestedStatement.pdf.map { pdf =>
+      PdfLink(
+        downloadURL = pdf.downloadURL,
+        formattedSize = pdf.formattedSize,
+        ariaText = messages(
+          "cf.security-statements.requested.download-link.aria-text",
+          Pdf,
+          dateAsDayMonthAndYear(requestedStatement.startDate),
+          dateAsDayMonthAndYear(requestedStatement.endDate),
+          pdf.formattedSize
+        )
+      )
+    }
+
+    StatementRow(
+      historyIndex = statementIndex,
+      eori,
+      date,
+      pdf,
+      rowId = s"requested-statements-list-$statementIndex-row-$requestedIndex",
+      dateCellId = s"requested-statements-list-$statementIndex-row-$requestedIndex-date-cell",
+      linkCellId = s"requested-statements-list-$statementIndex-row-$requestedIndex-link-cell"
+    )
+  }
+
+  def hasSecurityStatementsForEori(securityStatements: Seq[SecurityStatementsForEori]): Boolean = {
```

Comment:
Please make this and other methods (renderEoriHeading, renderPdfLink and renderMissingDocumentsGuidance) private. In tests, make use of ViewModel members' value